pyccel/codegen/printing/ccode.py

```diff
@@ -1020,7 +1032,17 @@ def _print_For(self, expr):
 
     def _print_CodeBlock(self, expr):
         body = []
+        allocs = set()
+        self._allocs.append(allocs)
         for b in expr.body :
+            if isinstance(b, Allocate):
+                allocs.add(b.variable)
+            if isinstance(b, AliasAssign):
+                if isinstance(b.rhs, IndexedElement):
+                    for i in b.rhs.args:
+                        if isinstance(i, Slice):
+                            allocs.add(b.lhs)
+                            break
```

Comment:
Can you explain why this needs to be done in the code block rather than in `_print_Allocate` and `_print_AliasAssign`?

In addition this doesn't look like a c specific problem to me. If it was well implemented with a `Deallocate` node then this could also be used in fortran or other languages we may wish to add in the future

Reply:
i put it there so i can keep track of which CodeBlock i am working on.